apollo-portal 1.8.2 接入 oidc 认证失败

<!-- 这段文字不会显示在你的内容中。为了避免重复的信息，方便后续的检索，在提issue之前，请检查如下事项。如果是比较新手级别的问题，推荐到讨论区https://github.com/ctripcorp/apollo/discussions 提问 -->

- [ ] 我已经检查过[discussions](https://github.com/ctripcorp/apollo/discussions)
- [ ] 我已经搜索过[issues](https://github.com/ctripcorp/apollo/issues)
- [ ] 我已经仔细检查过[FAQ](https://www.apolloconfig.com/#/zh/faq/common-issues-in-deployment-and-development-phase)

**描述bug**

apollo-portal 1.8.2 版本，使用 oidc 认证，报错不能启动；

**期望**

apollo-portal 1.8.2 能正常使用 oidc 认证

**截图**

![image](https://user-images.githubusercontent.com/20253863/121780147-a0b8c880-cbd1-11eb-971a-1b969cf15acd.png)


### 额外的细节和日志

- 版本：1.8.2
- 错误日志： 详见上面截图
- 配置：使用 k8s 部署；按照文档 https://www.apolloconfig.com/#/zh/development/portal-how-to-implement-user-login-function?id=%e5%ae%9e%e7%8e%b0%e6%96%b9%e5%bc%8f%e4%b8%89%ef%bc%9a-%e6%8e%a5%e5%85%a5oidc 参考，新增 application-oidc.yml ，该文件内容如下:
```yaml
spring:
  security:
    oauth2:
      client:
        provider:
          provider-name: test-openid
          issuer-uri: <openid 服务器 URL，不包含 .well-known/openid-configuration>
        registration:
          registration-name: apollo-portal
          authorization-grant-type: authorization_code
          client-authentication-method: basic
          client-id: <申请的客户端 ID>
          provider: test-openid
          scope:
            - openid
          client-secret: <与客户端ID 对应的 secret>
```
- 平台和操作系统：k8s 1.17.4

### 使用 authing 是为了测试，不用去额外处理敏感信息；现在我使用公司内部的 openid 服务来跑这个 demo，你看下是不是与 authing 同样的问题？ 另外，这个 demo 中有没有指定 alg 是 RS256 或 HS256 ，或者能否通过变量的形式传入指定的 alg ?
### 另外在 application.yml 中指定了 spring.security.oauth2.resourceserver.jwt.jws-algorithm = HS256，感觉没用；

### 使用公司内部的 openid 服务测试日志如下：
``` yaml
2021-06-17 10:59:32.831 DEBUG 22918 --- [qtp199657303-17] o.s.security.web.FilterChainProxy        : Securing GET /echo/text/123
2021-06-17 10:59:32.844 DEBUG 22918 --- [qtp199657303-17] s.s.w.c.SecurityContextPersistenceFilter : Set SecurityContextHolder to empty SecurityContext
2021-06-17 10:59:32.851 DEBUG 22918 --- [qtp199657303-17] o.s.s.w.a.AnonymousAuthenticationFilter  : Set SecurityContextHolder to anonymous SecurityContext
2021-06-17 10:59:32.867 DEBUG 22918 --- [qtp199657303-17] o.s.s.w.a.i.FilterSecurityInterceptor    : Failed to authorize filter invocation [GET /echo/text/123] with attributes [authenticated]
2021-06-17 10:59:32.918 DEBUG 22918 --- [qtp199657303-17] o.s.s.w.s.HttpSessionRequestCache        : Saved request http://<demo 所在服务器IP>:8070/echo/text/123 to session
2021-06-17 10:59:32.920 DEBUG 22918 --- [qtp199657303-17] s.w.a.DelegatingAuthenticationEntryPoint : Trying to match using And [Not [RequestHeaderRequestMatcher [expectedHeaderName=X-Requested-With, expectedHeaderValue=XMLHttpRequest]], MediaTypeRequestMatcher [contentNegotiationStrategy=org.springframework.web.accept.ContentNegotiationManager@34675dc8, matchingMediaTypes=[application/xhtml+xml, image/*, text/html, text/plain], useEquals=false, ignoredMediaTypes=[*/*]]]
2021-06-17 10:59:32.920 DEBUG 22918 --- [qtp199657303-17] s.w.a.DelegatingAuthenticationEntryPoint : Match found! Executing org.springframework.security.web.authentication.DelegatingAuthenticationEntryPoint@7f34a967
2021-06-17 10:59:32.923 DEBUG 22918 --- [qtp199657303-17] s.w.a.DelegatingAuthenticationEntryPoint : Trying to match using And [Not [RequestHeaderRequestMatcher [expectedHeaderName=X-Requested-With, expectedHeaderValue=XMLHttpRequest]], Not [And [Or [Ant [pattern='/login'], Ant [pattern='/favicon.ico']], And [Not [RequestHeaderRequestMatcher [expectedHeaderName=X-Requested-With, expectedHeaderValue=XMLHttpRequest]], MediaTypeRequestMatcher [contentNegotiationStrategy=org.springframework.web.accept.ContentNegotiationManager@34675dc8, matchingMediaTypes=[application/xhtml+xml, image/*, text/html, text/plain], useEquals=false, ignoredMediaTypes=[*/*]]]]]]
2021-06-17 10:59:32.923 DEBUG 22918 --- [qtp199657303-17] s.w.a.DelegatingAuthenticationEntryPoint : Match found! Executing org.springframework.security.web.authentication.LoginUrlAuthenticationEntryPoint@e1b72d
2021-06-17 10:59:32.924 DEBUG 22918 --- [qtp199657303-17] o.s.s.web.DefaultRedirectStrategy        : Redirecting to http://<demo 所在服务器IP>:8070/oauth2/authorization/authing
2021-06-17 10:59:32.925 DEBUG 22918 --- [qtp199657303-17] w.c.HttpSessionSecurityContextRepository : Did not store empty SecurityContext
2021-06-17 10:59:32.939 DEBUG 22918 --- [qtp199657303-17] w.c.HttpSessionSecurityContextRepository : Did not store empty SecurityContext
2021-06-17 10:59:32.942 DEBUG 22918 --- [qtp199657303-17] s.s.w.c.SecurityContextPersistenceFilter : Cleared SecurityContextHolder to complete request
2021-06-17 10:59:33.001 DEBUG 22918 --- [qtp199657303-12] o.s.security.web.FilterChainProxy        : Securing GET /oauth2/authorization/authing
2021-06-17 10:59:33.002 DEBUG 22918 --- [qtp199657303-12] s.s.w.c.SecurityContextPersistenceFilter : Set SecurityContextHolder to empty SecurityContext
2021-06-17 10:59:33.018 DEBUG 22918 --- [qtp199657303-12] o.s.s.web.DefaultRedirectStrategy        : Redirecting to https://login.test.com/connect/authorize?response_type=code&client_id=a8a38d34cf1711ebbfb9246e965dfd84&scope=openid%20profile%20email&state=-nkDLZCEo3m8CFF3AAbfFetePELE_x4Ocx67hcQApQ8%3D&redirect_uri=http://<demo 所在服务器IP>:8070/login/oauth2/code/authing&nonce=6qpDARVBT0ebgwQMszLsm4Kyw8-r1Z6dt2QHqfNDEcE
2021-06-17 10:59:33.018 DEBUG 22918 --- [qtp199657303-12] w.c.HttpSessionSecurityContextRepository : Did not store empty SecurityContext
2021-06-17 10:59:33.018 DEBUG 22918 --- [qtp199657303-12] w.c.HttpSessionSecurityContextRepository : Did not store empty SecurityContext
2021-06-17 10:59:33.019 DEBUG 22918 --- [qtp199657303-12] s.s.w.c.SecurityContextPersistenceFilter : Cleared SecurityContextHolder to complete request
2021-06-17 10:59:40.764 DEBUG 22918 --- [qtp199657303-16] o.s.security.web.FilterChainProxy        : Securing GET /login/oauth2/code/authing?state=-nkDLZCEo3m8CFF3AAbfFetePELE_x4Ocx67hcQApQ8%3D&code=1ccc9e84b8c446b2b918de53b73fe4d3
2021-06-17 10:59:40.765 DEBUG 22918 --- [qtp199657303-16] s.s.w.c.SecurityContextPersistenceFilter : Set SecurityContextHolder to empty SecurityContext
2021-06-17 10:59:40.784 DEBUG 22918 --- [qtp199657303-16] o.s.web.client.RestTemplate              : HTTP POST https://login.test.com/connect/token
2021-06-17 10:59:40.791 DEBUG 22918 --- [qtp199657303-16] o.s.web.client.RestTemplate              : Accept=[application/json, application/*+json]
2021-06-17 10:59:40.792 DEBUG 22918 --- [qtp199657303-16] o.s.web.client.RestTemplate              : Writing [{grant_type=[authorization_code], code=[1ccc9e84b8c446b2b918de53b73fe4d3], redirect_uri=[http://<demo 所在服务器IP>:8070/login/oauth2/code/authing]}] as "application/x-www-form-urlencoded;charset=UTF-8"
2021-06-17 10:59:40.990 DEBUG 22918 --- [qtp199657303-16] o.s.web.client.RestTemplate              : Response 200 OK
2021-06-17 10:59:40.995 DEBUG 22918 --- [qtp199657303-16] o.s.web.client.RestTemplate              : Reading to [org.springframework.security.oauth2.core.endpoint.OAuth2AccessTokenResponse] as "application/json"
2021-06-17 10:59:41.101 DEBUG 22918 --- [qtp199657303-16] .s.a.DefaultAuthenticationEventPublisher : No event was found for the exception org.springframework.security.oauth2.core.OAuth2AuthenticationException
2021-06-17 10:59:41.114 ERROR 22918 --- [qtp199657303-16] o.v.o.d.OidcWebSecurityConfigurerAdapter : authentication failure

org.springframework.security.oauth2.core.OAuth2AuthenticationException: [invalid_id_token] An error occurred while attempting to decode the Jwt: Signed JWT rejected: Another algorithm expected, or no matching key(s) found
	at org.springframework.security.oauth2.client.oidc.authentication.OidcAuthorizationCodeAuthenticationProvider.getJwt(OidcAuthorizationCodeAuthenticationProvider.java:249) ~[spring-security-oauth2-client-5.4.5.jar!/:5.4.5]
	at org.springframework.security.oauth2.client.oidc.authentication.OidcAuthorizationCodeAuthenticationProvider.createOidcToken(OidcAuthorizationCodeAuthenticationProvider.java:236) ~[spring-security-oauth2-client-5.4.5.jar!/:5.4.5]
	at org.springframework.security.oauth2.client.oidc.authentication.OidcAuthorizationCodeAuthenticationProvider.authenticate(OidcAuthorizationCodeAuthenticationProvider.java:154) ~[spring-security-oauth2-client-5.4.5.jar!/:5.4.5]
	at org.springframework.security.authentication.ProviderManager.authenticate(ProviderManager.java:182) ~[spring-security-core-5.4.5.jar!/:5.4.5]
	at org.springframework.security.oauth2.client.web.OAuth2LoginAuthenticationFilter.attemptAuthentication(OAuth2LoginAuthenticationFilter.java:192) ~[spring-security-oauth2-client-5.4.5.jar!/:5.4.5]
	at org.springframework.security.web.authentication.AbstractAuthenticationProcessingFilter.doFilter(AbstractAuthenticationProcessingFilter.java:222) ~[spring-security-web-5.4.5.jar!/:5.4.5]
	at org.springframework.security.web.authentication.AbstractAuthenticationProcessingFilter.doFilter(AbstractAuthenticationProcessingFilter.java:212) ~[spring-security-web-5.4.5.jar!/:5.4.5]
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:336) ~[spring-security-web-5.4.5.jar!/:5.4.5]
	at org.springframework.security.oauth2.client.web.OAuth2AuthorizationRequestRedirectFilter.doFilterInternal(OAuth2AuthorizationRequestRedirectFilter.java:178) ~[spring-security-oauth2-client-5.4.5.jar!/:5.4.5]
	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:119) ~[spring-web-5.3.4.jar!/:5.3.4]
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:336) ~[spring-security-web-5.4.5.jar!/:5.4.5]
	at org.springframework.security.oauth2.client.web.OAuth2AuthorizationRequestRedirectFilter.doFilterInternal(OAuth2AuthorizationRequestRedirectFilter.java:178) ~[spring-security-oauth2-client-5.4.5.jar!/:5.4.5]
	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:119) ~[spring-web-5.3.4.jar!/:5.3.4]
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:336) ~[spring-security-web-5.4.5.jar!/:5.4.5]
	at org.springframework.security.web.authentication.logout.LogoutFilter.doFilter(LogoutFilter.java:103) ~[spring-security-web-5.4.5.jar!/:5.4.5]
	at org.springframework.security.web.authentication.logout.LogoutFilter.doFilter(LogoutFilter.java:89) ~[spring-security-web-5.4.5.jar!/:5.4.5]
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:336) ~[spring-security-web-5.4.5.jar!/:5.4.5]
	at org.springframework.security.web.header.HeaderWriterFilter.doHeadersAfter(HeaderWriterFilter.java:90) ~[spring-security-web-5.4.5.jar!/:5.4.5]
	at org.springframework.security.web.header.HeaderWriterFilter.doFilterInternal(HeaderWriterFilter.java:75) ~[spring-security-web-5.4.5.jar!/:5.4.5]
	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:119) ~[spring-web-5.3.4.jar!/:5.3.4]
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:336) ~[spring-security-web-5.4.5.jar!/:5.4.5]
	at org.springframework.security.web.context.SecurityContextPersistenceFilter.doFilter(SecurityContextPersistenceFilter.java:110) ~[spring-security-web-5.4.5.jar!/:5.4.5]
	at org.springframework.security.web.context.SecurityContextPersistenceFilter.doFilter(SecurityContextPersistenceFilter.java:80) ~[spring-security-web-5.4.5.jar!/:5.4.5]
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:336) ~[spring-security-web-5.4.5.jar!/:5.4.5]
	at org.springframework.security.web.context.request.async.WebAsyncManagerIntegrationFilter.doFilterInternal(WebAsyncManagerIntegrationFilter.java:55) ~[spring-security-web-5.4.5.jar!/:5.4.5]
	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:119) ~[spring-web-5.3.4.jar!/:5.3.4]
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:336) ~[spring-security-web-5.4.5.jar!/:5.4.5]
	at org.springframework.security.web.FilterChainProxy.doFilterInternal(FilterChainProxy.java:211) ~[spring-security-web-5.4.5.jar!/:5.4.5]
	at org.springframework.security.web.FilterChainProxy.doFilter(FilterChainProxy.java:183) ~[spring-security-web-5.4.5.jar!/:5.4.5]
	at org.springframework.web.filter.DelegatingFilterProxy.invokeDelegate(DelegatingFilterProxy.java:358) ~[spring-web-5.3.4.jar!/:5.3.4]
	at org.springframework.web.filter.DelegatingFilterProxy.doFilter(DelegatingFilterProxy.java:271) ~[spring-web-5.3.4.jar!/:5.3.4]
	at org.eclipse.jetty.servlet.FilterHolder.doFilter(FilterHolder.java:193) ~[jetty-servlet-9.4.36.v20210114.jar!/:9.4.36.v20210114]
	at org.eclipse.jetty.servlet.ServletHandler$Chain.doFilter(ServletHandler.java:1601) ~[jetty-servlet-9.4.36.v20210114.jar!/:9.4.36.v20210114]
	at org.springframework.web.filter.RequestContextFilter.doFilterInternal(RequestContextFilter.java:100) ~[spring-web-5.3.4.jar!/:5.3.4]
	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:119) ~[spring-web-5.3.4.jar!/:5.3.4]
	at org.eclipse.jetty.servlet.FilterHolder.doFilter(FilterHolder.java:193) ~[jetty-servlet-9.4.36.v20210114.jar!/:9.4.36.v20210114]
	at org.eclipse.jetty.servlet.ServletHandler$Chain.doFilter(ServletHandler.java:1601) ~[jetty-servlet-9.4.36.v20210114.jar!/:9.4.36.v20210114]
	at org.springframework.web.filter.FormContentFilter.doFilterInternal(FormContentFilter.java:93) ~[spring-web-5.3.4.jar!/:5.3.4]
	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:119) ~[spring-web-5.3.4.jar!/:5.3.4]
	at org.eclipse.jetty.servlet.FilterHolder.doFilter(FilterHolder.java:193) ~[jetty-servlet-9.4.36.v20210114.jar!/:9.4.36.v20210114]
	at org.eclipse.jetty.servlet.ServletHandler$Chain.doFilter(ServletHandler.java:1601) ~[jetty-servlet-9.4.36.v20210114.jar!/:9.4.36.v20210114]
	at org.springframework.web.filter.CharacterEncodingFilter.doFilterInternal(CharacterEncodingFilter.java:201) ~[spring-web-5.3.4.jar!/:5.3.4]
	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:119) ~[spring-web-5.3.4.jar!/:5.3.4]
	at org.eclipse.jetty.servlet.FilterHolder.doFilter(FilterHolder.java:193) ~[jetty-servlet-9.4.36.v20210114.jar!/:9.4.36.v20210114]
	at org.eclipse.jetty.servlet.ServletHandler$Chain.doFilter(ServletHandler.java:1601) ~[jetty-servlet-9.4.36.v20210114.jar!/:9.4.36.v20210114]
	at org.eclipse.jetty.servlet.ServletHandler.doHandle(ServletHandler.java:548) ~[jetty-servlet-9.4.36.v20210114.jar!/:9.4.36.v20210114]
	at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:143) ~[jetty-server-9.4.36.v20210114.jar!/:9.4.36.v20210114]
	at org.eclipse.jetty.security.SecurityHandler.handle(SecurityHandler.java:602) ~[jetty-security-9.4.36.v20210114.jar!/:9.4.36.v20210114]
	at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:127) ~[jetty-server-9.4.36.v20210114.jar!/:9.4.36.v20210114]
	at org.eclipse.jetty.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:235) ~[jetty-server-9.4.36.v20210114.jar!/:9.4.36.v20210114]
	at org.eclipse.jetty.server.session.SessionHandler.doHandle(SessionHandler.java:1624) ~[jetty-server-9.4.36.v20210114.jar!/:9.4.36.v20210114]
	at org.eclipse.jetty.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:233) ~[jetty-server-9.4.36.v20210114.jar!/:9.4.36.v20210114]
	at org.eclipse.jetty.server.handler.ContextHandler.doHandle(ContextHandler.java:1435) ~[jetty-server-9.4.36.v20210114.jar!/:9.4.36.v20210114]
	at org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:188) ~[jetty-server-9.4.36.v20210114.jar!/:9.4.36.v20210114]
	at org.eclipse.jetty.servlet.ServletHandler.doScope(ServletHandler.java:501) ~[jetty-servlet-9.4.36.v20210114.jar!/:9.4.36.v20210114]
	at org.eclipse.jetty.server.session.SessionHandler.doScope(SessionHandler.java:1594) ~[jetty-server-9.4.36.v20210114.jar!/:9.4.36.v20210114]
	at org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:186) ~[jetty-server-9.4.36.v20210114.jar!/:9.4.36.v20210114]
	at org.eclipse.jetty.server.handler.ContextHandler.doScope(ContextHandler.java:1350) ~[jetty-server-9.4.36.v20210114.jar!/:9.4.36.v20210114]
	at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:141) ~[jetty-server-9.4.36.v20210114.jar!/:9.4.36.v20210114]
	at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:127) ~[jetty-server-9.4.36.v20210114.jar!/:9.4.36.v20210114]
	at org.eclipse.jetty.server.Server.handle(Server.java:516) ~[jetty-server-9.4.36.v20210114.jar!/:9.4.36.v20210114]
	at org.eclipse.jetty.server.HttpChannel.lambda$handle$1(HttpChannel.java:388) ~[jetty-server-9.4.36.v20210114.jar!/:9.4.36.v20210114]
	at org.eclipse.jetty.server.HttpChannel.dispatch(HttpChannel.java:633) ~[jetty-server-9.4.36.v20210114.jar!/:9.4.36.v20210114]
	at org.eclipse.jetty.server.HttpChannel.handle(HttpChannel.java:380) ~[jetty-server-9.4.36.v20210114.jar!/:9.4.36.v20210114]
	at org.eclipse.jetty.server.HttpConnection.onFillable(HttpConnection.java:273) ~[jetty-server-9.4.36.v20210114.jar!/:9.4.36.v20210114]
	at org.eclipse.jetty.io.AbstractConnection$ReadCallback.succeeded(AbstractConnection.java:311) ~[jetty-io-9.4.36.v20210114.jar!/:9.4.36.v20210114]
	at org.eclipse.jetty.io.FillInterest.fillable(FillInterest.java:105) ~[jetty-io-9.4.36.v20210114.jar!/:9.4.36.v20210114]
	at org.eclipse.jetty.io.ChannelEndPoint$1.run(ChannelEndPoint.java:104) ~[jetty-io-9.4.36.v20210114.jar!/:9.4.36.v20210114]
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.runTask(EatWhatYouKill.java:336) ~[jetty-util-9.4.36.v20210114.jar!/:9.4.36.v20210114]
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.doProduce(EatWhatYouKill.java:313) ~[jetty-util-9.4.36.v20210114.jar!/:9.4.36.v20210114]
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.tryProduce(EatWhatYouKill.java:171) ~[jetty-util-9.4.36.v20210114.jar!/:9.4.36.v20210114]
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.run(EatWhatYouKill.java:129) ~[jetty-util-9.4.36.v20210114.jar!/:9.4.36.v20210114]
	at org.eclipse.jetty.util.thread.ReservedThreadExecutor$ReservedThread.run(ReservedThreadExecutor.java:375) ~[jetty-util-9.4.36.v20210114.jar!/:9.4.36.v20210114]
	at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:773) ~[jetty-util-9.4.36.v20210114.jar!/:9.4.36.v20210114]
	at org.eclipse.jetty.util.thread.QueuedThreadPool$Runner.run(QueuedThreadPool.java:905) ~[jetty-util-9.4.36.v20210114.jar!/:9.4.36.v20210114]
	at java.base/java.lang.Thread.run(Thread.java:829) ~[na:na]
Caused by: org.springframework.security.oauth2.jwt.BadJwtException: An error occurred while attempting to decode the Jwt: Signed JWT rejected: Another algorithm expected, or no matching key(s) found
	at org.springframework.security.oauth2.jwt.NimbusJwtDecoder.createJwt(NimbusJwtDecoder.java:179) ~[spring-security-oauth2-jose-5.4.5.jar!/:5.4.5]
	at org.springframework.security.oauth2.jwt.NimbusJwtDecoder.decode(NimbusJwtDecoder.java:136) ~[spring-security-oauth2-jose-5.4.5.jar!/:5.4.5]
	at org.springframework.security.oauth2.client.oidc.authentication.OidcAuthorizationCodeAuthenticationProvider.getJwt(OidcAuthorizationCodeAuthenticationProvider.java:245) ~[spring-security-oauth2-client-5.4.5.jar!/:5.4.5]
	... 75 common frames omitted
Caused by: com.nimbusds.jose.proc.BadJOSEException: Signed JWT rejected: Another algorithm expected, or no matching key(s) found
	at com.nimbusds.jwt.proc.DefaultJWTProcessor.process(DefaultJWTProcessor.java:384) ~[nimbus-jose-jwt-8.20.2.jar!/:8.20.2]
	at com.nimbusds.jwt.proc.DefaultJWTProcessor.process(DefaultJWTProcessor.java:330) ~[nimbus-jose-jwt-8.20.2.jar!/:8.20.2]
	at org.springframework.security.oauth2.jwt.NimbusJwtDecoder.createJwt(NimbusJwtDecoder.java:153) ~[spring-security-oauth2-jose-5.4.5.jar!/:5.4.5]
	... 77 common frames omitted

2021-06-17 10:59:41.120 ERROR 22918 --- [qtp199657303-16] o.v.o.d.OidcWebSecurityConfigurerAdapter : oauth2 authentication failure [[invalid_id_token] An error occurred while attempting to decode the Jwt: Signed JWT rejected: Another algorithm expected, or no matching key(s) found]
2021-06-17 10:59:41.121 DEBUG 22918 --- [qtp199657303-16] .a.SimpleUrlAuthenticationFailureHandler : Sending 401 Unauthorized error
2021-06-17 10:59:41.121 DEBUG 22918 --- [qtp199657303-16] w.c.HttpSessionSecurityContextRepository : Did not store empty SecurityContext
2021-06-17 10:59:41.121 DEBUG 22918 --- [qtp199657303-16] w.c.HttpSessionSecurityContextRepository : Did not store empty SecurityContext
2021-06-17 10:59:41.121 DEBUG 22918 --- [qtp199657303-16] s.s.w.c.SecurityContextPersistenceFilter : Cleared SecurityContextHolder to complete request
2021-06-17 10:59:41.130 DEBUG 22918 --- [qtp199657303-16] o.s.security.web.FilterChainProxy        : Securing GET /error?state=-nkDLZCEo3m8CFF3AAbfFetePELE_x4Ocx67hcQApQ8%3D&code=1ccc9e84b8c446b2b918de53b73fe4d3
2021-06-17 10:59:41.131 DEBUG 22918 --- [qtp199657303-16] s.s.w.c.SecurityContextPersistenceFilter : Set SecurityContextHolder to empty SecurityContext
2021-06-17 10:59:41.131 DEBUG 22918 --- [qtp199657303-16] o.s.s.w.a.AnonymousAuthenticationFilter  : Set SecurityContextHolder to anonymous SecurityContext
2021-06-17 10:59:41.135 DEBUG 22918 --- [qtp199657303-16] o.s.s.w.a.i.FilterSecurityInterceptor    : Failed to authorize filter invocation [GET /error?state=-nkDLZCEo3m8CFF3AAbfFetePELE_x4Ocx67hcQApQ8%3D&code=1ccc9e84b8c446b2b918de53b73fe4d3] with attributes [authenticated]
2021-06-17 10:59:41.138 DEBUG 22918 --- [qtp199657303-16] o.s.s.w.s.HttpSessionRequestCache        : Saved request http://<demo 所在服务器IP>:8070/error?state=-nkDLZCEo3m8CFF3AAbfFetePELE_x4Ocx67hcQApQ8%3D&code=1ccc9e84b8c446b2b918de53b73fe4d3 to session
2021-06-17 10:59:41.138 DEBUG 22918 --- [qtp199657303-16] s.w.a.DelegatingAuthenticationEntryPoint : Trying to match using And [Not [RequestHeaderRequestMatcher [expectedHeaderName=X-Requested-With, expectedHeaderValue=XMLHttpRequest]], MediaTypeRequestMatcher [contentNegotiationStrategy=org.springframework.web.accept.ContentNegotiationManager@34675dc8, matchingMediaTypes=[application/xhtml+xml, image/*, text/html, text/plain], useEquals=false, ignoredMediaTypes=[*/*]]]
2021-06-17 10:59:41.138 DEBUG 22918 --- [qtp199657303-16] s.w.a.DelegatingAuthenticationEntryPoint : Match found! Executing org.springframework.security.web.authentication.DelegatingAuthenticationEntryPoint@7f34a967
2021-06-17 10:59:41.139 DEBUG 22918 --- [qtp199657303-16] s.w.a.DelegatingAuthenticationEntryPoint : Trying to match using And [Not [RequestHeaderRequestMatcher [expectedHeaderName=X-Requested-With, expectedHeaderValue=XMLHttpRequest]], Not [And [Or [Ant [pattern='/login'], Ant [pattern='/favicon.ico']], And [Not [RequestHeaderRequestMatcher [expectedHeaderName=X-Requested-With, expectedHeaderValue=XMLHttpRequest]], MediaTypeRequestMatcher [contentNegotiationStrategy=org.springframework.web.accept.ContentNegotiationManager@34675dc8, matchingMediaTypes=[application/xhtml+xml, image/*, text/html, text/plain], useEquals=false, ignoredMediaTypes=[*/*]]]]]]
2021-06-17 10:59:41.139 DEBUG 22918 --- [qtp199657303-16] s.w.a.DelegatingAuthenticationEntryPoint : Match found! Executing org.springframework.security.web.authentication.LoginUrlAuthenticationEntryPoint@e1b72d
2021-06-17 10:59:41.139 DEBUG 22918 --- [qtp199657303-16] o.s.s.web.DefaultRedirectStrategy        : Redirecting to http://<demo 所在服务器IP>:8070/oauth2/authorization/authing
2021-06-17 10:59:41.139 DEBUG 22918 --- [qtp199657303-16] w.c.HttpSessionSecurityContextRepository : Did not store empty SecurityContext
2021-06-17 10:59:41.139 DEBUG 22918 --- [qtp199657303-16] w.c.HttpSessionSecurityContextRepository : Did not store empty SecurityContext
2021-06-17 10:59:41.139 DEBUG 22918 --- [qtp199657303-16] s.s.w.c.SecurityContextPersistenceFilter : Cleared SecurityContextHolder to complete request
2021-06-17 10:59:41.202 DEBUG 22918 --- [qtp199657303-18] o.s.security.web.FilterChainProxy        : Securing GET /oauth2/authorization/authing
2021-06-17 10:59:41.204 DEBUG 22918 --- [qtp199657303-18] s.s.w.c.SecurityContextPersistenceFilter : Set SecurityContextHolder to empty SecurityContext
2021-06-17 10:59:41.205 DEBUG 22918 --- [qtp199657303-18] o.s.s.web.DefaultRedirectStrategy        : Redirecting to https://login.test.com/connect/authorize?response_type=code&client_id=a8a38d34cf1711ebbfb9246e965dfd84&scope=openid%20profile%20email&state=yHad7wTOdRf0JgHZezzZzrHO9hTsI2ASVtPODRhIBGQ%3D&redirect_uri=http://<demo 所在服务器IP>:8070/login/oauth2/code/authing&nonce=y_Il5Yl9uEU8HBiCxNQQBYKUe30mDl-uZAGpy4NPRzk
2021-06-17 10:59:41.205 DEBUG 22918 --- [qtp199657303-18] w.c.HttpSessionSecurityContextRepository : Did not store empty SecurityContext
2021-06-17 10:59:41.206 DEBUG 22918 --- [qtp199657303-18] w.c.HttpSessionSecurityContextRepository : Did not store empty SecurityContext
2021-06-17 10:59:41.206 DEBUG 22918 --- [qtp199657303-18] s.s.w.c.SecurityContextPersistenceFilter : Cleared SecurityContextHolder to complete request
2021-06-17 11:00:38.127 DEBUG 22918 --- [qtp199657303-14] o.s.security.web.FilterChainProxy        : Securing GET /login/oauth2/code/authing?state=yHad7wTOdRf0JgHZezzZzrHO9hTsI2ASVtPODRhIBGQ%3D&code=8001708477b346298caa1ea2e02e2abb
2021-06-17 11:00:38.127 DEBUG 22918 --- [qtp199657303-14] s.s.w.c.SecurityContextPersistenceFilter : Set SecurityContextHolder to empty SecurityContext
2021-06-17 11:00:38.133 DEBUG 22918 --- [qtp199657303-14] o.s.web.client.RestTemplate              : HTTP POST https://login.test.com/connect/token
2021-06-17 11:00:38.133 DEBUG 22918 --- [qtp199657303-14] o.s.web.client.RestTemplate              : Accept=[application/json, application/*+json]
2021-06-17 11:00:38.134 DEBUG 22918 --- [qtp199657303-14] o.s.web.client.RestTemplate              : Writing [{grant_type=[authorization_code], code=[8001708477b346298caa1ea2e02e2abb], redirect_uri=[http://<demo 所在服务器IP>:8070/login/oauth2/code/authing]}] as "application/x-www-form-urlencoded;charset=UTF-8"
2021-06-17 11:00:38.317 DEBUG 22918 --- [qtp199657303-14] o.s.web.client.RestTemplate              : Response 200 OK
2021-06-17 11:00:38.318 DEBUG 22918 --- [qtp199657303-14] o.s.web.client.RestTemplate              : Reading to [org.springframework.security.oauth2.core.endpoint.OAuth2AccessTokenResponse] as "application/json"
2021-06-17 11:00:38.321 DEBUG 22918 --- [qtp199657303-14] .s.a.DefaultAuthenticationEventPublisher : No event was found for the exception org.springframework.security.oauth2.core.OAuth2AuthenticationException
2021-06-17 11:00:38.328 ERROR 22918 --- [qtp199657303-14] o.v.o.d.OidcWebSecurityConfigurerAdapter : authentication failure

org.springframework.security.oauth2.core.OAuth2AuthenticationException: [invalid_id_token] An error occurred while attempting to decode the Jwt: Signed JWT rejected: Another algorithm expected, or no matching key(s) found
	at org.springframework.security.oauth2.client.oidc.authentication.OidcAuthorizationCodeAuthenticationProvider.getJwt(OidcAuthorizationCodeAuthenticationProvider.java:249) ~[spring-security-oauth2-client-5.4.5.jar!/:5.4.5]
	at org.springframework.security.oauth2.client.oidc.authentication.OidcAuthorizationCodeAuthenticationProvider.createOidcToken(OidcAuthorizationCodeAuthenticationProvider.java:236) ~[spring-security-oauth2-client-5.4.5.jar!/:5.4.5]
	at org.springframework.security.oauth2.client.oidc.authentication.OidcAuthorizationCodeAuthenticationProvider.authenticate(OidcAuthorizationCodeAuthenticationProvider.java:154) ~[spring-security-oauth2-client-5.4.5.jar!/:5.4.5]
	at org.springframework.security.authentication.ProviderManager.authenticate(ProviderManager.java:182) ~[spring-security-core-5.4.5.jar!/:5.4.5]
	at org.springframework.security.oauth2.client.web.OAuth2LoginAuthenticationFilter.attemptAuthentication(OAuth2LoginAuthenticationFilter.java:192) ~[spring-security-oauth2-client-5.4.5.jar!/:5.4.5]
	at org.springframework.security.web.authentication.AbstractAuthenticationProcessingFilter.doFilter(AbstractAuthenticationProcessingFilter.java:222) ~[spring-security-web-5.4.5.jar!/:5.4.5]
	at org.springframework.security.web.authentication.AbstractAuthenticationProcessingFilter.doFilter(AbstractAuthenticationProcessingFilter.java:212) ~[spring-security-web-5.4.5.jar!/:5.4.5]
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:336) ~[spring-security-web-5.4.5.jar!/:5.4.5]
	at org.springframework.security.oauth2.client.web.OAuth2AuthorizationRequestRedirectFilter.doFilterInternal(OAuth2AuthorizationRequestRedirectFilter.java:178) ~[spring-security-oauth2-client-5.4.5.jar!/:5.4.5]
	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:119) ~[spring-web-5.3.4.jar!/:5.3.4]
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:336) ~[spring-security-web-5.4.5.jar!/:5.4.5]
	at org.springframework.security.oauth2.client.web.OAuth2AuthorizationRequestRedirectFilter.doFilterInternal(OAuth2AuthorizationRequestRedirectFilter.java:178) ~[spring-security-oauth2-client-5.4.5.jar!/:5.4.5]
	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:119) ~[spring-web-5.3.4.jar!/:5.3.4]
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:336) ~[spring-security-web-5.4.5.jar!/:5.4.5]
	at org.springframework.security.web.authentication.logout.LogoutFilter.doFilter(LogoutFilter.java:103) ~[spring-security-web-5.4.5.jar!/:5.4.5]
	at org.springframework.security.web.authentication.logout.LogoutFilter.doFilter(LogoutFilter.java:89) ~[spring-security-web-5.4.5.jar!/:5.4.5]
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:336) ~[spring-security-web-5.4.5.jar!/:5.4.5]
	at org.springframework.security.web.header.HeaderWriterFilter.doHeadersAfter(HeaderWriterFilter.java:90) ~[spring-security-web-5.4.5.jar!/:5.4.5]
	at org.springframework.security.web.header.HeaderWriterFilter.doFilterInternal(HeaderWriterFilter.java:75) ~[spring-security-web-5.4.5.jar!/:5.4.5]
	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:119) ~[spring-web-5.3.4.jar!/:5.3.4]
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:336) ~[spring-security-web-5.4.5.jar!/:5.4.5]
	at org.springframework.security.web.context.SecurityContextPersistenceFilter.doFilter(SecurityContextPersistenceFilter.java:110) ~[spring-security-web-5.4.5.jar!/:5.4.5]
	at org.springframework.security.web.context.SecurityContextPersistenceFilter.doFilter(SecurityContextPersistenceFilter.java:80) ~[spring-security-web-5.4.5.jar!/:5.4.5]
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:336) ~[spring-security-web-5.4.5.jar!/:5.4.5]
	at org.springframework.security.web.context.request.async.WebAsyncManagerIntegrationFilter.doFilterInternal(WebAsyncManagerIntegrationFilter.java:55) ~[spring-security-web-5.4.5.jar!/:5.4.5]
	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:119) ~[spring-web-5.3.4.jar!/:5.3.4]
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:336) ~[spring-security-web-5.4.5.jar!/:5.4.5]
	at org.springframework.security.web.FilterChainProxy.doFilterInternal(FilterChainProxy.java:211) ~[spring-security-web-5.4.5.jar!/:5.4.5]
	at org.springframework.security.web.FilterChainProxy.doFilter(FilterChainProxy.java:183) ~[spring-security-web-5.4.5.jar!/:5.4.5]
	at org.springframework.web.filter.DelegatingFilterProxy.invokeDelegate(DelegatingFilterProxy.java:358) ~[spring-web-5.3.4.jar!/:5.3.4]
	at org.springframework.web.filter.DelegatingFilterProxy.doFilter(DelegatingFilterProxy.java:271) ~[spring-web-5.3.4.jar!/:5.3.4]
	at org.eclipse.jetty.servlet.FilterHolder.doFilter(FilterHolder.java:193) ~[jetty-servlet-9.4.36.v20210114.jar!/:9.4.36.v20210114]
	at org.eclipse.jetty.servlet.ServletHandler$Chain.doFilter(ServletHandler.java:1601) ~[jetty-servlet-9.4.36.v20210114.jar!/:9.4.36.v20210114]
	at org.springframework.web.filter.RequestContextFilter.doFilterInternal(RequestContextFilter.java:100) ~[spring-web-5.3.4.jar!/:5.3.4]
	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:119) ~[spring-web-5.3.4.jar!/:5.3.4]
	at org.eclipse.jetty.servlet.FilterHolder.doFilter(FilterHolder.java:193) ~[jetty-servlet-9.4.36.v20210114.jar!/:9.4.36.v20210114]
	at org.eclipse.jetty.servlet.ServletHandler$Chain.doFilter(ServletHandler.java:1601) ~[jetty-servlet-9.4.36.v20210114.jar!/:9.4.36.v20210114]
	at org.springframework.web.filter.FormContentFilter.doFilterInternal(FormContentFilter.java:93) ~[spring-web-5.3.4.jar!/:5.3.4]
	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:119) ~[spring-web-5.3.4.jar!/:5.3.4]
	at org.eclipse.jetty.servlet.FilterHolder.doFilter(FilterHolder.java:193) ~[jetty-servlet-9.4.36.v20210114.jar!/:9.4.36.v20210114]
	at org.eclipse.jetty.servlet.ServletHandler$Chain.doFilter(ServletHandler.java:1601) ~[jetty-servlet-9.4.36.v20210114.jar!/:9.4.36.v20210114]
	at org.springframework.web.filter.CharacterEncodingFilter.doFilterInternal(CharacterEncodingFilter.java:201) ~[spring-web-5.3.4.jar!/:5.3.4]
	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:119) ~[spring-web-5.3.4.jar!/:5.3.4]
	at org.eclipse.jetty.servlet.FilterHolder.doFilter(FilterHolder.java:193) ~[jetty-servlet-9.4.36.v20210114.jar!/:9.4.36.v20210114]
	at org.eclipse.jetty.servlet.ServletHandler$Chain.doFilter(ServletHandler.java:1601) ~[jetty-servlet-9.4.36.v20210114.jar!/:9.4.36.v20210114]
	at org.eclipse.jetty.servlet.ServletHandler.doHandle(ServletHandler.java:548) ~[jetty-servlet-9.4.36.v20210114.jar!/:9.4.36.v20210114]
	at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:143) ~[jetty-server-9.4.36.v20210114.jar!/:9.4.36.v20210114]
	at org.eclipse.jetty.security.SecurityHandler.handle(SecurityHandler.java:602) ~[jetty-security-9.4.36.v20210114.jar!/:9.4.36.v20210114]
	at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:127) ~[jetty-server-9.4.36.v20210114.jar!/:9.4.36.v20210114]
	at org.eclipse.jetty.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:235) ~[jetty-server-9.4.36.v20210114.jar!/:9.4.36.v20210114]
	at org.eclipse.jetty.server.session.SessionHandler.doHandle(SessionHandler.java:1624) ~[jetty-server-9.4.36.v20210114.jar!/:9.4.36.v20210114]
	at org.eclipse.jetty.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:233) ~[jetty-server-9.4.36.v20210114.jar!/:9.4.36.v20210114]
	at org.eclipse.jetty.server.handler.ContextHandler.doHandle(ContextHandler.java:1435) ~[jetty-server-9.4.36.v20210114.jar!/:9.4.36.v20210114]
	at org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:188) ~[jetty-server-9.4.36.v20210114.jar!/:9.4.36.v20210114]
	at org.eclipse.jetty.servlet.ServletHandler.doScope(ServletHandler.java:501) ~[jetty-servlet-9.4.36.v20210114.jar!/:9.4.36.v20210114]
	at org.eclipse.jetty.server.session.SessionHandler.doScope(SessionHandler.java:1594) ~[jetty-server-9.4.36.v20210114.jar!/:9.4.36.v20210114]
	at org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:186) ~[jetty-server-9.4.36.v20210114.jar!/:9.4.36.v20210114]
	at org.eclipse.jetty.server.handler.ContextHandler.doScope(ContextHandler.java:1350) ~[jetty-server-9.4.36.v20210114.jar!/:9.4.36.v20210114]
	at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:141) ~[jetty-server-9.4.36.v20210114.jar!/:9.4.36.v20210114]
	at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:127) ~[jetty-server-9.4.36.v20210114.jar!/:9.4.36.v20210114]
	at org.eclipse.jetty.server.Server.handle(Server.java:516) ~[jetty-server-9.4.36.v20210114.jar!/:9.4.36.v20210114]
	at org.eclipse.jetty.server.HttpChannel.lambda$handle$1(HttpChannel.java:388) ~[jetty-server-9.4.36.v20210114.jar!/:9.4.36.v20210114]
	at org.eclipse.jetty.server.HttpChannel.dispatch(HttpChannel.java:633) ~[jetty-server-9.4.36.v20210114.jar!/:9.4.36.v20210114]
	at org.eclipse.jetty.server.HttpChannel.handle(HttpChannel.java:380) ~[jetty-server-9.4.36.v20210114.jar!/:9.4.36.v20210114]
	at org.eclipse.jetty.server.HttpConnection.onFillable(HttpConnection.java:273) ~[jetty-server-9.4.36.v20210114.jar!/:9.4.36.v20210114]
	at org.eclipse.jetty.io.AbstractConnection$ReadCallback.succeeded(AbstractConnection.java:311) ~[jetty-io-9.4.36.v20210114.jar!/:9.4.36.v20210114]
	at org.eclipse.jetty.io.FillInterest.fillable(FillInterest.java:105) ~[jetty-io-9.4.36.v20210114.jar!/:9.4.36.v20210114]
	at org.eclipse.jetty.io.ChannelEndPoint$1.run(ChannelEndPoint.java:104) ~[jetty-io-9.4.36.v20210114.jar!/:9.4.36.v20210114]
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.runTask(EatWhatYouKill.java:336) ~[jetty-util-9.4.36.v20210114.jar!/:9.4.36.v20210114]
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.doProduce(EatWhatYouKill.java:313) ~[jetty-util-9.4.36.v20210114.jar!/:9.4.36.v20210114]
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.tryProduce(EatWhatYouKill.java:171) ~[jetty-util-9.4.36.v20210114.jar!/:9.4.36.v20210114]
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.run(EatWhatYouKill.java:129) ~[jetty-util-9.4.36.v20210114.jar!/:9.4.36.v20210114]
	at org.eclipse.jetty.util.thread.ReservedThreadExecutor$ReservedThread.run(ReservedThreadExecutor.java:375) ~[jetty-util-9.4.36.v20210114.jar!/:9.4.36.v20210114]
	at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:773) ~[jetty-util-9.4.36.v20210114.jar!/:9.4.36.v20210114]
	at org.eclipse.jetty.util.thread.QueuedThreadPool$Runner.run(QueuedThreadPool.java:905) ~[jetty-util-9.4.36.v20210114.jar!/:9.4.36.v20210114]
	at java.base/java.lang.Thread.run(Thread.java:829) ~[na:na]
Caused by: org.springframework.security.oauth2.jwt.BadJwtException: An error occurred while attempting to decode the Jwt: Signed JWT rejected: Another algorithm expected, or no matching key(s) found
	at org.springframework.security.oauth2.jwt.NimbusJwtDecoder.createJwt(NimbusJwtDecoder.java:179) ~[spring-security-oauth2-jose-5.4.5.jar!/:5.4.5]
	at org.springframework.security.oauth2.jwt.NimbusJwtDecoder.decode(NimbusJwtDecoder.java:136) ~[spring-security-oauth2-jose-5.4.5.jar!/:5.4.5]
	at org.springframework.security.oauth2.client.oidc.authentication.OidcAuthorizationCodeAuthenticationProvider.getJwt(OidcAuthorizationCodeAuthenticationProvider.java:245) ~[spring-security-oauth2-client-5.4.5.jar!/:5.4.5]
	... 75 common frames omitted
Caused by: com.nimbusds.jose.proc.BadJOSEException: Signed JWT rejected: Another algorithm expected, or no matching key(s) found
	at com.nimbusds.jwt.proc.DefaultJWTProcessor.process(DefaultJWTProcessor.java:384) ~[nimbus-jose-jwt-8.20.2.jar!/:8.20.2]
	at com.nimbusds.jwt.proc.DefaultJWTProcessor.process(DefaultJWTProcessor.java:330) ~[nimbus-jose-jwt-8.20.2.jar!/:8.20.2]
	at org.springframework.security.oauth2.jwt.NimbusJwtDecoder.createJwt(NimbusJwtDecoder.java:153) ~[spring-security-oauth2-jose-5.4.5.jar!/:5.4.5]
	... 77 common frames omitted

2021-06-17 11:00:38.334 ERROR 22918 --- [qtp199657303-14] o.v.o.d.OidcWebSecurityConfigurerAdapter : oauth2 authentication failure [[invalid_id_token] An error occurred while attempting to decode the Jwt: Signed JWT rejected: Another algorithm expected, or no matching key(s) found]
2021-06-17 11:00:38.334 DEBUG 22918 --- [qtp199657303-14] .a.SimpleUrlAuthenticationFailureHandler : Sending 401 Unauthorized error
2021-06-17 11:00:38.334 DEBUG 22918 --- [qtp199657303-14] w.c.HttpSessionSecurityContextRepository : Did not store empty SecurityContext
2021-06-17 11:00:38.334 DEBUG 22918 --- [qtp199657303-14] w.c.HttpSessionSecurityContextRepository : Did not store empty SecurityContext
2021-06-17 11:00:38.335 DEBUG 22918 --- [qtp199657303-14] s.s.w.c.SecurityContextPersistenceFilter : Cleared SecurityContextHolder to complete request
2021-06-17 11:00:38.335 DEBUG 22918 --- [qtp199657303-14] o.s.security.web.FilterChainProxy        : Securing GET /error?state=yHad7wTOdRf0JgHZezzZzrHO9hTsI2ASVtPODRhIBGQ%3D&code=8001708477b346298caa1ea2e02e2abb
2021-06-17 11:00:38.336 DEBUG 22918 --- [qtp199657303-14] s.s.w.c.SecurityContextPersistenceFilter : Set SecurityContextHolder to empty SecurityContext
2021-06-17 11:00:38.336 DEBUG 22918 --- [qtp199657303-14] o.s.s.w.a.AnonymousAuthenticationFilter  : Set SecurityContextHolder to anonymous SecurityContext
2021-06-17 11:00:38.336 DEBUG 22918 --- [qtp199657303-14] o.s.s.w.a.i.FilterSecurityInterceptor    : Failed to authorize filter invocation [GET /error?state=yHad7wTOdRf0JgHZezzZzrHO9hTsI2ASVtPODRhIBGQ%3D&code=8001708477b346298caa1ea2e02e2abb] with attributes [authenticated]
2021-06-17 11:00:38.339 DEBUG 22918 --- [qtp199657303-14] o.s.s.w.s.HttpSessionRequestCache        : Saved request http://<demo 所在服务器IP>:8070/error?state=yHad7wTOdRf0JgHZezzZzrHO9hTsI2ASVtPODRhIBGQ%3D&code=8001708477b346298caa1ea2e02e2abb to session
2021-06-17 11:00:38.339 DEBUG 22918 --- [qtp199657303-14] s.w.a.DelegatingAuthenticationEntryPoint : Trying to match using And [Not [RequestHeaderRequestMatcher [expectedHeaderName=X-Requested-With, expectedHeaderValue=XMLHttpRequest]], MediaTypeRequestMatcher [contentNegotiationStrategy=org.springframework.web.accept.ContentNegotiationManager@34675dc8, matchingMediaTypes=[application/xhtml+xml, image/*, text/html, text/plain], useEquals=false, ignoredMediaTypes=[*/*]]]
2021-06-17 11:00:38.340 DEBUG 22918 --- [qtp199657303-14] s.w.a.DelegatingAuthenticationEntryPoint : Match found! Executing org.springframework.security.web.authentication.DelegatingAuthenticationEntryPoint@7f34a967
2021-06-17 11:00:38.340 DEBUG 22918 --- [qtp199657303-14] s.w.a.DelegatingAuthenticationEntryPoint : Trying to match using And [Not [RequestHeaderRequestMatcher [expectedHeaderName=X-Requested-With, expectedHeaderValue=XMLHttpRequest]], Not [And [Or [Ant [pattern='/login'], Ant [pattern='/favicon.ico']], And [Not [RequestHeaderRequestMatcher [expectedHeaderName=X-Requested-With, expectedHeaderValue=XMLHttpRequest]], MediaTypeRequestMatcher [contentNegotiationStrategy=org.springframework.web.accept.ContentNegotiationManager@34675dc8, matchingMediaTypes=[application/xhtml+xml, image/*, text/html, text/plain], useEquals=false, ignoredMediaTypes=[*/*]]]]]]
2021-06-17 11:00:38.340 DEBUG 22918 --- [qtp199657303-14] s.w.a.DelegatingAuthenticationEntryPoint : Match found! Executing org.springframework.security.web.authentication.LoginUrlAuthenticationEntryPoint@e1b72d
2021-06-17 11:00:38.340 DEBUG 22918 --- [qtp199657303-14] o.s.s.web.DefaultRedirectStrategy        : Redirecting to http://<demo 所在服务器IP>:8070/oauth2/authorization/authing
2021-06-17 11:00:38.342 DEBUG 22918 --- [qtp199657303-14] w.c.HttpSessionSecurityContextRepository : Did not store empty SecurityContext
2021-06-17 11:00:38.344 DEBUG 22918 --- [qtp199657303-14] w.c.HttpSessionSecurityContextRepository : Did not store empty SecurityContext
2021-06-17 11:00:38.345 DEBUG 22918 --- [qtp199657303-14] s.s.w.c.SecurityContextPersistenceFilter : Cleared SecurityContextHolder to complete request
2021-06-17 11:00:38.399 DEBUG 22918 --- [qtp199657303-17] o.s.security.web.FilterChainProxy        : Securing GET /oauth2/authorization/authing
2021-06-17 11:00:38.399 DEBUG 22918 --- [qtp199657303-17] s.s.w.c.SecurityContextPersistenceFilter : Set SecurityContextHolder to empty SecurityContext
2021-06-17 11:00:38.402 DEBUG 22918 --- [qtp199657303-17] o.s.s.web.DefaultRedirectStrategy        : Redirecting to https://login.test.com/connect/authorize?response_type=code&client_id=a8a38d34cf1711ebbfb9246e965dfd84&scope=openid%20profile%20email&state=hlbzqsh1WnDbuy8p-Ijt452gJZi4R_vtYTWgLoVryVk%3D&redirect_uri=http://<demo 所在服务器IP>:8070/login/oauth2/code/authing&nonce=M7Ggo__QKBQgn869K6OOA-SlTiXga--okfkl6g0Sc9s
2021-06-17 11:00:38.407 DEBUG 22918 --- [qtp199657303-17] w.c.HttpSessionSecurityContextRepository : Did not store empty SecurityContext
2021-06-17 11:00:38.408 DEBUG 22918 --- [qtp199657303-17] w.c.HttpSessionSecurityContextRepository : Did not store empty SecurityContext
2021-06-17 11:00:38.410 DEBUG 22918 --- [qtp199657303-17] s.s.w.c.SecurityContextPersistenceFilter : Cleared SecurityContextHolder to complete request
```

--- a/CHANGES.md
+++ b/CHANGES.md
@@ -45,6 +45,7 @@ Apollo 1.9.0
 * [fix oidc sql](https://github.com/ctripcorp/apollo/pull/3720)
 * [feat(apollo-client): add method interestedChangedKeys to ConfigChangeEvent](https://github.com/ctripcorp/apollo/pull/3666)
 * [add More... link for known users](https://github.com/ctripcorp/apollo/pull/3757)
+* [update OIDC documentation](https://github.com/ctripcorp/apollo/pull/3766)
 ------------------
 All issues and pull requests are [here](https://github.com/ctripcorp/apollo/milestone/6?closed=1)
 

--- a/docs/zh/development/portal-how-to-implement-user-login-function.md
+++ b/docs/zh/development/portal-how-to-implement-user-login-function.md
@@ -232,7 +232,7 @@ export JAVA_OPTS="$JAVA_OPTS -Dspring.profiles.active=github,ldap"
 从 1.8.0 版本开始支持 OpenID Connect 登录, 这种实现方式的前提是已经部署了 OpenID Connect 登录服务  
 配置前需要准备:
 * OpenID Connect 的提供者配置端点(符合 RFC 8414 标准的 issuer-uri), 需要是 **https** 的, 例如 https://host:port/auth/realms/apollo/.well-known/openid-configuration
-* 在 OpenID Connect 服务里创建一个 client, 获取 client-id 以及对应的 client-secret
+* 在 OpenID Connect 服务里创建一个 client, idToken 的签名算法必须设置为 **RS256**, 获取 client-id 以及对应的 client-secret
 
 ### 1. 配置 `application-oidc.yml`
 

